config/src/test/resources/org/springframework/security/config/http/ClientRegistrationsBeanDefinitionParserTests-MultiClientRegistration.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2002-2020 the original author or authors.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~       https://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<b:beans xmlns:b="http://www.springframework.org/schema/beans"
+		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		xmlns="http://www.springframework.org/schema/security"
+		xsi:schemaLocation="
+			http://www.springframework.org/schema/security
+			https://www.springframework.org/schema/security/spring-security.xsd
+			http://www.springframework.org/schema/beans
+			https://www.springframework.org/schema/beans/spring-beans.xsd">
+	<client-registrations>
```

Comment:
Instead of duplicating `google` and `github` `<client-registration>` in the test xml configs, let's use `<b:import resource="registration.xml"/>` instead. I would suggest creating `google-registration.xml`, `github-registration.xml` and `google-github-registration.xml`. You can then `<b:import resource=../>` in the xml configs.

Reply:
I had applied the `<b:import resource=../>` in the test xml configs. there is a caveat in using import for this parser.

this is related with issue gh-5347 auto-redirect when only one client configured
See `OAuth2LoginBeanDefinitionParser.getLoginEntryPoint`
`Element clientRegsElt = DomUtils.getChildElementByTagName(element.getOwnerDocument().getDocumentElement(),Elements.CLIENT_REGISTRATIONS);`

It won't be able to find `client-registrations` element if this element was not configured in the same XML file.

and as we know there is now way to be able to get/check the contents of `ClientRegistrationRepository` bean in this parser.

Reply:
Just want to update, i just found a way to get `ApplicationContext` and check for the `ClientRegistrationRepository` in which i applied it on the `loginLinks` (see my latest changes soon) but this will still be a problem, see method `AuthenticationConfigBuilder.selectEntryPoint` couldn't check `ApplicationContext` on that part